### PR TITLE
Remove underscores in adf name and change prefix -> suffix

### LINKED ISSUE #43: Resource naming should be consistent with AzureTRE/UCLH conventions
Azure deployed resources in the TRE follow a
```
{resource abbreviation}-{tre_id}
```
convention for the most part. While the UCLH guidance is
```
{Resource Type}-{Org Name}-{Region}-{Environment}-{Workload}
```

I suggest we at least put the resource abbreviation first where possible. This means the naming prefix should become a naming suffix

### LINKED ISSUE #48: ADF name cannot have underscores

### PR DESCRIPTION
**Pull request by t-young31**
### Fixes #43

### CODE REVIEW

**Comment by t-young31** [comment]
can't try a deploy here as the PR bot is using the workflow on main so:

Screenshot 2023-01-31 at 14 57 07

**Comment by t-young31** [comment]
Deploy on my fork is ok: https://github.com/t-young31/FlowEHR/actions/runs/4055612259

File: infrastructure/transform/main.tf [hunk 1]
```diff
@@ -57,7 +57,7 @@ resource "azurerm_role_assignment" "adf_can_create_clusters" {
 }
 
 resource "azurerm_data_factory" "adf" {
-  name                = "${var.naming_prefix}-adf"
+  name                = replace("adf-${var.naming_suffix}", "_", "")
```
Comment (jjgriff93) [inline infrastructure/transform/main.tf:60]: I'm thinking perhaps we do away with `_` altogether and stipulate this in the `config.schema`, or we replace with `""`? There's a bit too much variation to account for with the current approach

**Review by jjgriff93 — APPROVED** [review]

File: scripts/name_suffix.py [hunk 2]
```diff
@@ -40,11 +32,14 @@ def _remove_non_alpha_numeric_chars(string: str) -> str:
 def naming_suffix():
     """
     Construct a naming suffix that satisfies the naming requirements for a resource
-    group. Any hyphens in $SUFFIX and $ENVIRONMENT to separate "blocks"
+    group. Any hyphens, underscores or spaces in $SUFFIX and $ENVIRONMENT will be delted
```
Comment (jjgriff93) [inline scripts/name_suffix.py:35]: deleted

**Pull request merged**